### PR TITLE
Fix build error in tls_client_main.c

### PR DESCRIPTION
**Pull request by dr-venkman**

### CODE REVIEW

**Comment by dr-venkman** [comment]
@pillip8282 Could you please review the fix, and merge if found ok? Thank you.

File: apps/examples/tls_client/tls_client_main.c [hunk 1]
```diff
@@ -462,7 +462,7 @@ int tls_client_cb(void *args)
 	int tail_len;
 	int i;
 	int written;
-	int frags;
+	int frags = 0;
```
Comment (lokeshbv) [inline apps/examples/tls_client/tls_client_main.c]: could you please mention, what build error.

File: apps/examples/websocket/websocket_main.c [hunk 2]
```diff
@@ -795,10 +795,6 @@ int websocket_main(int argc, char *argv[])
 		goto error_with_function_failure;
 	}
 
-	if (input) {
-		free(input);
-	}
-
 	return 0;
```
Comment (lokeshbv) [inline apps/examples/websocket/websocket_main.c]: what if it never goes to the `error_with_function_failure:` label.?

File: apps/examples/websocket/websocket_main.c [hunk 3]
```diff
@@ -795,10 +795,6 @@ int websocket_main(int argc, char *argv[])
 		goto error_with_function_failure;
 	}
```
Comment (pradeepns85) [inline apps/examples/websocket/websocket_main.c]: Kindly release the allocated memory for "input" during success case as well before returning.

Comment (dr-venkman) [reply apps/examples/websocket/websocket_main.c]: That is exactly what we should not be doing, hence the commit. "input"is passed as a parameter to both websocket_client and websocket_server, and both these functions free input (referred to as 'opt' inside). So calling free to input after success should be avoided.

File: apps/examples/websocket/websocket_main.c (hunk 2 above)
Comment (dr-venkman) [reply apps/examples/websocket/websocket_main.c]: As mentioned above, if the functions above are executed, "input" will be freed. So in the case of success, it is not necessary to free input. Please let me know if the reasoning is correct.

File: apps/examples/tls_client/tls_client_main.c (hunk 1 above)
Comment (dr-venkman) [reply apps/examples/tls_client/tls_client_main.c]: Uninitialized variable warning treated as error.

**Comment by pradeepns85** [comment]
Hi.

I don't think websocket_client will always free the input ! .   There are so many error scenarios in which it returns from websocket_client function, without freeing the input. so There is a possibility that input will become a dangling pointer !!  

444 /* websocket client prints received messages from a server using recv message cb */
445 int websocket_client(void *arg)
446 {

476         if (g_wcenabled) {
477                 printf("\nWebsocket client is already running\n");
478                 return WEBSOCKET_INIT_ERROR;
479         }
480         g_wcenabled = 1;
481 
482         if (size < 16) {
483                 printf("wrong size\n %s\n", WEBSOCKET_USAGE);
484                 g_wcenabled = 0;
485                 return -1;
486         }
487         if (send_cnt < 1) {
488                 printf("wrong send count\n %s\n", WEBSOCKET_USAGE);
489                 g_wcenabled = 0;
490                 return -1;
491         }
}

I hope you got my point.

Thanks,
Pradeep
Thanks,
Pradeep

File: apps/examples/websocket/websocket_main.c (hunk 3 above)
Comment (davidfather) [reply apps/examples/websocket/websocket_main.c]: @dr-venkman If both websocket_client and websocket_server are racing to free 'input', what happens?

**Pull request merged**